### PR TITLE
Avoid refetch in `RealmLive` when operating strictly on the update thread

### PR DESCRIPTION
**Pull request by peppy**
Second attempt at this (see https://github.com/ppy/osu/pull/16498). Now with better safeties and statistics.

Previous benchmarks still apply.

### CODE REVIEW

File: osu.Game/Database/RealmLive.cs [hunk 1]
```diff
@@ -125,4 +165,12 @@ private T retrieveFromID(Realm realm, Guid id)
 
         public override string ToString() => PerformRead(i => i.ToString());
     }
+
+    public class RealmLive
```
Comment (smoogipoo) [inline osu.Game/Database/RealmLive.cs:169]: Maybe a static class? Named `RealmLiveStatistics`? I don't like that this is being inherited from and constructable.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Comment by smoogipoo** [comment]
You say "better safeties", but is there something that prompted this? No tests have been added or anything. Put another way, what's "unsafe" about the current implementation and why is that not tested against?

**Comment by peppy** [comment]
I meant that the reason I closed the previous PR has now been resolved (the re-fetch was sometimes failing due to lack of refresh). See discussion from [this post](https://github.com/ppy/osu/pull/16498#issuecomment-1017864168) onwards.

**Review by smoogipoo — APPROVED** [review]
Looks fine I suppose...

**Pull request merged**